File: apps/android/app/src/main/java/com/barns/app/presentation/myitems/ProductItemPresentation.kt
package com.barns.app.presentation.myitems

import com.barns.app.domain.model.ProductItem
import com.barns.app.domain.model.ProductItemStatus
import com.barns.app.domain.model.ProductItemType

/**
 * Presentation-only display helper that frames a registered [ProductItem] as
 * the customer's own/installed greenery — closer to a "My Appliances for
 * greenery" registry than a generic care list. Parity with the iOS
 * ProductItemPresentation introduced in PR #27.
 *
 * This is a pure value type derived from existing domain fields only. It adds
 * no new domain model, no persistence, and no data access; it just shapes
 * calm, support-oriented copy for the My Items list and detail screens.
 */
data class ProductItemPresentation(
    val name: String,
    val categoryLabel: String,
    val typeLabel: String,
    val locationLabel: String,
    val statusLabel: String,
    val careStatusLabel: String,
    val nextActionHint: String,
    val ownershipSummary: String,
) {
    companion object {
        fun from(item: ProductItem): ProductItemPresentation {
            val careLinked = item.careGuideIds.isNotEmpty()
            return ProductItemPresentation(
                name = item.name,
                categoryLabel = categoryLabel(item.categoryId),
                typeLabel = typeLabel(item.type),
                locationLabel = item.locationLabel ?: "Location not set",
                statusLabel = when (item.status) {
                    ProductItemStatus.ACTIVE -> "In your care"
                    ProductItemStatus.ARCHIVED -> "Archived"
                },
                careStatusLabel = if (careLinked) {
                    "Care guidance linked"
                } else {
                    "No care guide linked yet"
                },
                nextActionHint = if (careLinked) {
                    "Open care guidance for your next step"
                } else {
                    "Browse care guidance to set up routine care"
                },
                ownershipSummary = ownershipSummary(item.type),
            )
        }

        private fun categoryLabel(categoryId: String): String = when (categoryId) {
            "cat-wall-green" -> "Wall greenery"
            "cat-interior-green" -> "Interior greenery"
            else -> {
                // Fall back to a readable label derived from the id, e.g.
                // "cat-roof-garden" -> "Roof garden".
                val trimmed = categoryId.removePrefix("cat-")
                val words = trimmed.split("-").filter { it.isNotEmpty() }
                if (words.isEmpty()) {
                    "Greenery"
                } else {
                    val first = words.first().replaceFirstChar { it.uppercase() }
                    val rest = words.drop(1).joinToString(" ")
                    if (rest.isEmpty()) first else "$first $rest"
                }
            }
        }

        private fun typeLabel(type: ProductItemType): String = when (type) {
            ProductItemType.INSTALLED -> "Installed greenery"
            ProductItemType.PURCHASED -> "Owned greenery"
        }

        private fun ownershipSummary(type: ProductItemType): String = when (type) {
            ProductItemType.INSTALLED -> "Installed greenery in your care"
            ProductItemType.PURCHASED -> "Greenery you own"
        }
    }
}

File: apps/ios/Barns/BarnsApp/Presentation/MyItems/ProductItemPresentation.swift
import Foundation

/// Presentation-only display helper that frames a registered `ProductItem`
/// as the customer's own/installed greenery — closer to a "My Appliances for
/// greenery" registry than a generic care list.
///
/// This is a pure value type derived from existing domain fields only. It adds
/// no new domain model, no persistence, and no data access; it just shapes
/// calm, support-oriented copy for the My Items list and detail screens.
struct ProductItemPresentation: Equatable {
    let name: String
    let categoryLabel: String
    let typeLabel: String
    let locationLabel: String
    let statusLabel: String
    let careStatusLabel: String
    let nextActionHint: String
    let ownershipSummary: String

    init(item: ProductItem) {
        name = item.name
        categoryLabel = Self.makeCategoryLabel(for: item.categoryId)
        typeLabel = Self.makeTypeLabel(for: item.type)
        locationLabel = item.locationLabel ?? "Location not set"

        switch item.status {
        case .active:
            statusLabel = "In your care"
        case .archived:
            statusLabel = "Archived"
        }

        if item.careGuideIds.isEmpty {
            careStatusLabel = "No care guide linked yet"
            nextActionHint = "Browse care guidance to set up routine care"
        } else {
            careStatusLabel = "Care guidance linked"
            nextActionHint = "Open care guidance for your next step"
        }

        ownershipSummary = Self.makeOwnershipSummary(for: item.type)
    }

    private static func makeCategoryLabel(for categoryId: String) -> String {
        switch categoryId {
        case "cat-wall-green":
            return "Wall greenery"
        case "cat-interior-green":
            return "Interior greenery"
        default:
            // Fall back to a readable label derived from the id, e.g.
            // "cat-roof-garden" -> "Roof garden".
            let trimmed = categoryId.hasPrefix("cat-")
                ? String(categoryId.dropFirst("cat-".count))
                : categoryId
            let words = trimmed.split(separator: "-").map(String.init)
            guard let first = words.first, !first.isEmpty else { return "Greenery" }
            let rest = words.dropFirst().joined(separator: " ")
            let label = (first.capitalized + (rest.isEmpty ? "" : " " + rest))
            return label.isEmpty ? "Greenery" : label
        }
    }

    private static func makeTypeLabel(for type: ProductItemType) -> String {
        switch type {
        case .installed:
            return "Installed greenery"
        case .purchased:
            return "Owned greenery"
        }
    }

    private static func makeOwnershipSummary(for type: ProductItemType) -> String {
        switch type {
        case .installed:
            return "Installed greenery in your care"
        case .purchased:
            return "Greenery you own"
        }
    }
}
